### DISCUSSION TITLE
boost::pfr

### DISCUSSION THREAD

**Discussion by xR3b0rn** [boostorg/spirit#690] [Q&A] [answered]
Is there any reason why `boost::spriti::x3` is not using `boost::pfr`? If there are concerns about increasements in compiletime or if you do not want have this additional dependency, `boost::pfr` could be optional (e.g. deactivatable by macro).

**Comment by Kojoley** [comment, marked as answer]
At the moment sequence handling is built on top of Fusion and highly coupled to it. I was planning to introduce an intermediate layer that could be an extension point to utilize PFR, Hana, and any other tuple-like library.

**Reply by daminetreg** [reply]
Just as note in the meantime someone built it : https://github.com/boostorg/fusion/pull/241

**Reply by denzor200** [reply]
> I was planning to introduce an intermediate layer that could be an extension point to utilize PFR, Hana, and any other tuple-like library.

How are things going with it, is there anything moving on? May I help you with something?

**Reply by xR3b0rn** [reply]
There is a MR ready to merge here: https://github.com/boostorg/fusion/pull/266

However, there are no relevant updates since January.

**Reply by denzor200** [reply]
> There is a MR ready to merge here

Sure I know, I've created that PR :)

> However, there are no relevant updates since January.

That's the reason why I'm trying to ping here in the Spirit repo

**Reply by denzor200** [reply]
It's quite possible that Spirit side PFR integration will take less effort that Fusion side, why not?.
From my side: I'm ready to spend time to implement it, of course should I receive the confirmation that @Kojoley or someone else is not working on it already

**Reply by djowel** [reply]
I can chime in as the author of both (though I am not that active anymore). I haven't heard from @Kojoley lately, I hope everything is fine.  Meanwhile, can you please give me a summary of the current status, what's involved here, and what needs to be done? It's confusing to have information scattered all over the place.

**Reply by denzor200** [reply]
Hi, @djowel ! I hope everything fine too.
The current status - everything done on the Fusion side: https://github.com/boostorg/fusion/pull/266
Waiting for review for that PR. And of course it's possible to split the PR to 2 small ones if it's too much for anyone who will provide review. Or I can start working on the same functionality one the Spirit side, if nobody have enough motivation to maintain the Fusion at all.
Let me know what you think what should I do next.

**Reply by djowel** [reply]
OK, looking at https://github.com/boostorg/fusion/pull/266 seems we are waiting for a review from @daminetreg or someone else?

"@xR3b0rn everything ready from my side, waiting for review. As I informed above, I was unavailable for some months. But right now I'm available every day and can react to review pretty fast."

I'm not sure if I'll have the time to do a review. There's a lot! So yes, breaking it down into smaller PRs will help immensely. It would also be nice if you can ask for others (maybe from the boost list) to do a review?

**Reply by denzor200** [reply]
> OK, looking at https://github.com/boostorg/fusion/pull/266 seems we are waiting for a review from @daminetreg or someone else?

Yes, from @daminetreg 

> I'm not sure if I'll have the time to do a review. There's a lot! So yes, breaking it down into smaller PRs will help immensely. It would also be nice if you can ask for others (maybe from the boost list) to do a review?

Ok, then later I will split that PR and then ask someone from mail-list for review

**Reply by daminetreg** [reply]
Yes I apologize, I never came to it, always thought I can do this but was having a hard time making availabilities. I still want to use it and test it in codebases where I rely alot on BOOST_FUSION_ADAPT_STRUCT but am overbusy with work.

**Reply by denzor200** [reply]
Hi, Damien! Dont worry, everything going to be OK.

BTW, do you know that now PFR can reflect field names? :-)
